r1cs/gadgets/crypto/src/prf/ripemd160.rs

```diff
@@ -0,0 +1,458 @@
+//! Circuits for the [RIPEMD160] hash function
+//!
+//! [RIPEMD160]: https://tools.ietf.org/html/rfc6234
+
+use r1cs_std::boolean::Boolean;
+use r1cs_std::eq::MultiEq;
+use r1cs_std::uint32::UInt32;
+use r1cs_std::uint8::UInt8;
+use r1cs_core::{ConstraintSystem, SynthesisError};
+use algebra::PrimeField;
+
+use crate::sha256::{sha256_ch_boolean, triop};
+
+/// Outputs K[round_idx] and K'[round_idx]
+fn get_round_constants(round_idx: usize) -> (UInt32, UInt32) {
```

Comment:
Why do you need to define this as a function?
Wasn't it easier to define
const K : [usize; 80] = [0x00000000,...,0xa953fd4e];
const K_PRIME: ...

Reply:
Constants are the same for a range of round, so it would've been very ugly and long: I decided to use a function instead